File: Python/underautomation/universal_robots/rtde/rtde_data_package_event_args.py
import typing
from underautomation.universal_robots.rtde.rtde_output_values import RtdeOutputValues
from underautomation.universal_robots.common.package_event_args import PackageEventArgs
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..",  'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.Rtde import RtdeDataPackageEventArgs as rtde_data_package_event_args

class RtdeDataPackageEventArgs(PackageEventArgs):
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = rtde_data_package_event_args()
		else:
			self._instance = _internal
	@property
	def output_recipe_id(self) -> int:
		return self._instance.OutputRecipeId
	@output_recipe_id.setter
	def output_recipe_id(self, value: int):
		self._instance.OutputRecipeId = value
	@property
	def output_data_values(self) -> RtdeOutputValues:
		return RtdeOutputValues(self._instance.OutputDataValues)
	@output_data_values.setter
	def output_data_values(self, value: RtdeOutputValues):
		self._instance.OutputDataValues = value
	@property
	def measured_frequency(self) -> float:
		return self._instance.MeasuredFrequency
	@measured_frequency.setter
	def measured_frequency(self, value: float):
		self._instance.MeasuredFrequency = value

File: UnderAutomation.UniversalRobots.ObfuscatedSources/Rtde/RtdeDataPackageEventArgs.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.

using Common;

namespace Rtde {
	/// <summary>
	/// Event arguments for an RTDE output data package received from the robot at the subscribed frequency.
	/// </summary>
	public class RtdeDataPackageEventArgs : PackageEventArgs {


		public override string ToString()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}


		public RtdeDataPackageEventArgs()
		{
			// Source is hidden, a Source licence is needed to access internal code...
		}

		/// <summary>
		/// Gets or sets the recipe identifier for the received output data (RTDE v2 only; 0 for v1).
		/// </summary>
		public byte OutputRecipeId { get; set; }

		/// <summary>
		/// Gets or sets the decoded output values contained in this data package.
		/// </summary>
		public RtdeOutputValues OutputDataValues { get; set; }

		/// <summary>
		/// Gets or sets the measured frequency in Hz, computed from successive <xref href="UnderAutomation.UniversalRobots.Rtde.RtdeOutputData.Timestamp" data-throw-if-not-resolved="false"></xref> values.
		/// </summary>
		public double MeasuredFrequency { get; set; }
	}
}
